src/java.base/share/classes/sun/security/ssl/SSLConfiguration.java

```diff
@@ -469,31 +476,36 @@ public Object clone() {
 
         if (property != null && !property.isEmpty()) {
             String[] signatureSchemeNames = property.split(",");
-            List<SignatureScheme> signatureSchemes =
-                        new ArrayList<>(signatureSchemeNames.length);
-            for (int i = 0; i < signatureSchemeNames.length; i++) {
-                signatureSchemeNames[i] = signatureSchemeNames[i].trim();
-                if (signatureSchemeNames[i].isEmpty()) {
+            List<String> signatureSchemes =
+                    new ArrayList<>(signatureSchemeNames.length);
+            for (String schemeName : signatureSchemeNames) {
+                schemeName = schemeName.trim();
+                if (schemeName.isEmpty()) {
                     continue;
                 }
 
-                SignatureScheme scheme =
-                    SignatureScheme.nameOf(signatureSchemeNames[i]);
+                // Check the availability
+                SignatureScheme scheme = SignatureScheme.nameOf(schemeName);
                 if (scheme != null && scheme.isAvailable) {
-                    signatureSchemes.add(scheme);
+                    signatureSchemes.add(schemeName);
                 } else {
                     if (SSLLogger.isOn && SSLLogger.isOn("ssl,sslctx")) {
                         SSLLogger.fine(
-                                "The current installed providers do not " +
-                                "support signature scheme: " +
-                                signatureSchemeNames[i]);
+                        "The current installed providers do not " +
+                              "support signature scheme: " + schemeName);
                     }
                 }
             }
 
-            return signatureSchemes;
+            // Note that if the System Property value is not defined (JDK
+            // default value) or empty, the provider-specific default is used.
```

Comment:
I think you can remove this comment as it is repeated on lines 507-508 (and makes more sense there).